test/types/serverFactory.test-d.ts

```diff
@@ -0,0 +1,34 @@
+// /**
```

Comment:
Why is this file commented?

Reply:
The types should be valid, but the tests are failing. I commented it out so it could get landed and other parts could be worked on (such as the schema work, which I might add is going very well PR coming soon!). If you'd like for this to be solved before landing this PR then I understand and will re-prioritize it.

Reply:
I do not like landing code that is commented out. Either uncomment it and make it work, or open a tracking issue of this refactoring and make a note about this and then remove the file.

Reply:
Pushing a commit fixing this now 👍